packages/@headlessui-react/src/hooks/document-overflow/overflow-store.ts

```diff
@@ -53,16 +56,27 @@ export let overflows = createStore(() => new Map<Document, DocEntry>(), {
     if (entry) {
       entry.count--
       entry.meta.delete(meta)
+      entry.computedMeta = buildMeta(entry.meta)
     }
 
     return this
   },
 
-  SCROLL_PREVENT({ doc, d, meta }: DocEntry) {
+  SCROLL_PREVENT(entry: DocEntry) {
     let ctx = {
-      doc,
-      d,
-      meta: buildMeta(meta),
+      doc: entry.doc,
+      d: entry.d,
+
+      // The moment we `PUSH`, we also `SCROLL_PREVENT`. But a later `PUSH` will
+      // not re-trigger a `SCROLL_PREVENT` because we are already in a locked
+      // state.
+      //
+      // This `meta()` function is called lazily such that a `PUSH` or `POP`
+      // that happens later can update the meta information. Otherwise we would
+      // use stale meta information.
+      meta() {
+        return entry.computedMeta
```

Comment:
Used `entry` instead of destructuring the `computedMeta` to ensure we are looking at the latest data. Alternatively we could use `this.get(doc).computedMeta` or something

Reply:
seems fine 🤷‍♂️